src/System.ServiceModel.Primitives/tests/ServiceModel/ChannelFactoryTest.cs

```diff
@@ -200,4 +201,27 @@ public static void CreateChannel_Of_Typed_Proxy_Using_BasicHttpBinding()
             }
         }
     }
+
+    [Fact]
+    public static void ChannelFactory_Async_Open_Close() {
+        ChannelFactory<IRequestChannel> factory = null;
+        try {
+            BasicHttpBinding binding = new BasicHttpBinding();
+
+            // Create the channel factory
+            factory = new ChannelFactory<IRequestChannel>(binding, new EndpointAddress(FakeAddress.HttpAddress));
+            Assert.Equal(CommunicationState.Created, factory.State);
+
+            Task.Factory.FromAsync(factory.BeginOpen(null, null), factory.EndOpen).GetAwaiter().GetResult();
+            Assert.Equal(CommunicationState.Opened, factory.State);
```

Comment:
Nit: I might use Assert.True so you can provide a meaningful message to distinguish the 2 asserts.  Otherwise you won't know which one failed.  This is especially important to diagnose failure in a lab environment.